test/suite/test_debug_mode06.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2019 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+
+# test_debug_mode06.py
+#   Test the debug mode settings. Test slow_checkpoint use (from WT-4981).
+#   Note: testing timing will make results unreliable so we won't do that.
+class test_debug_mode06(wttest.WiredTigerTestCase):
+    conn_config = 'log=(enabled=true),debug_mode=(slow_checkpoint=true)'
+    uri = 'file:test_debug_mode06'
+
+    # Test (re)configuring the flag on and off.
+    def test_slow_checkpoints(self):
+        self.session.create(self.uri, 'key_format=s,value_format=s')
```

Comment:
While this tests that the API configuration string can be parsed, is there anything the test can do to verify it works as expected? For instance, instead of having two separate tests, have one test that might do something like this that tests both the API on/off and that it does what it is supposed to do:
* set the `conn_config` as you do above.
* create the table.
* insert a bunch of data into the table.
* perform a checkpoint and record the time it takes to do so.
* reconfigure the setting to false.
* insert the same amount of data into the table.
* perform a checkpoint and record the time it takes.
* assert time1 > time2.

Reply:
We did have some conversation about testing the actual functionality in Python and I recommended against it:
https://jira.mongodb.org/browse/WT-4921?focusedCommentId=2635468&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2635468
We've had a lot of difficulties getting timing related Python testing reliable.

Reply:
Perhaps I can add a few lines which just executes a checkpoint. That way it just ensures that having the flag set doesn't break existing functionality.

Reply:
Ok, I can agree with relative timings. How about this as a sort of middle ground that does not do comparison between checkpoints:
* insert a few items into the table (so checkpoint should actually be fast)
* perform a checkpoint with the setting on.
* get the last checkpoint time.
* assert it took longer than 10000 usecs or whatever the sleep unit it.

Reply:
The statistic I have in mind is `txn_checkpoint_time_recent`. So something like:
`ckpt_time = stat__cursor[stat.conn.txn_checkpoint_time_recent][2]`
`self.assertTrue(ckpt_time >= 10)`

(The `wt_sleep` unit is usecs and the stat unit is msecs.)

Reply:
I've added a simple insert of data + checkpoint just to make sure the flag doesn't affect functionality.